src/os/bluestore/NVMEDevice.cc

```diff
@@ -504,6 +503,8 @@ int NVMEManager::try_get(const string &sn_tag, SharedDriverData **driver)
   ProbeContext ctx = {sn_tag, this, nullptr};
   r = spdk_nvme_probe(&ctx, probe_cb, attach_cb);
   if (r || !ctx.driver) {
+    if (!ctx.driver)
+      r = -ENOENT;
```

Comment:
@yuyuyu101 Well, I fully understand what you mean. But theoretically there is one possibility we hit the below condition:
```
  r == 0 && ctx.driver == nullptr
```
which will also bring we here.

So I think the right logic should be(based on your suggestion):
```
if (r < 0) {
  assert(!ctx.driver);
}
```

What do you think?